### PR TITLE
[JENKINS-40909] Continue to extend AbstractStepExecutionImpl for serial form compatibility

### PR DESCRIPTION
**Pull request by jglick**
Corrects a regressiion in #23: [JENKINS-40909](https://issues.jenkins-ci.org/browse/JENKINS-40909)

Downstream of https://github.com/jenkinsci/workflow-step-api-plugin/pull/17 & https://github.com/jenkinsci/workflow-support-plugin/pull/28.

@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -79,23 +79,23 @@
...
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-support</artifactId>
-            <version>2.1</version>
+            <version>2.12-20170109.225138-3</version> <!-- TODO https://github.com/jenkinsci/workflow-support-plugin/pull/28 -->
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-cps</artifactId>
-            <version>2.6</version>
+            <version>2.22</version>
```
Comment (jglick) [inline pom.xml]: Not 2.23: https://github.com/jenkinsci/workflow-cps-plugin/pull/95

**Comment by abayer** [comment]
:bee:

**Comment by jglick** [comment]
> I don't follow the testcase 100% here

It is pretty hairy—the first time I managed to make a test reproduce an upgrade of a running durable task, and it took me hours to get the details right.

File: pom.xml [hunk 2]
```diff
@@ -107,16 +107,27 @@
...
+            <groupId>org.jenkins-ci.plugins.workflow</groupId>
+            <artifactId>workflow-scm-step</artifactId>
+            <version>2.3</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>script-security</artifactId>
+            <version>1.25</version>
+        </dependency>
```
Comment (recena) [inline pom.xml]: @jglick scope `test`?????

Comment (jglick) [inline pom.xml]: No, this was already an indirect dependency IIRC, just an older one.

Comment (recena) [inline pom.xml]: Ok!

**Pull request merged**